### DISCUSSION TITLE
Why do you need to set `NOTIFICATION_<type>` to enable a notification provider?

### DISCUSSION THREAD

**Discussion by Andre601** [upptime/upptime#653] [Q&A]
There is one configuration setup that I can't really wrap my head around, which is the one to enable the notification provider(s).

As an example, if I want to enable notifications through a GitHub Webhook, would I need to add the following secrets:

- `NOTIFICATION_DISCORD` with value `true`
- `NOTIFICATION_DISCORD_WEBHOOK` with value `true`
- `NOTIFICATION_DISCORD_WEBHOOK_URL` with the value being the Webhook URL.

I can't see a reason as to why I have to set two arbitrary values, just to have an option active here. Wouldn't it also work to check if `NOTIFICATION_DISCORD_WEBHOOK_URL` is present and not empty to then try to push a notification? Why the extra two options which do the exact same thing, which is to enable the setting to begin with?

This same strange setup is also present for Slack and Telegram and neither of these three options has another setting that would justify such a setup. If they had another `NOTIFICATION_<service>_<option>` then I could understand the purpose as it would allow you to define what specific option of the service should be used, but since neither of these three services has any additional option is this setup completely arbitrary.

**Comment by AnandChowdhary** [comment]
Yeah definitely annoying to have to add an extra enabler :)

The logic here was for example with emails, you're saying:

- `NOTIFICATION_EMAIL=true` tells us that you want to enable notifications via email
- `NOTIFICATION_EMAIL_SENDGRID=true` tells us that you want to use Sendgrid for emails
- `NOTIFICATION_EMAIL_SENDGRID_API_KEY=...` gives us the API key

The idea was that you can for example keep `NOTIFICATION_EMAIL_SENDGRID=true` but disable emails directly without having to remove Sendgrid configuration.

So to make it consistent, we added this to all, even in places where there are no subcategories. I agree that it makes a ton of sense that if we have the Sendgrid API key, we assume that you want to use Sendgrid and send emails.
